apps/tx-builder/src/components/EditTransactionModal.tsx

```diff
@@ -0,0 +1,128 @@
+  SolidityFormValuesTypes,
+  TO_ADDRESS_FIELD_NAME,
+} from './forms/SolidityForm';
+import { weiToEther } from '../utils';
+
+type EditTransactionModalProps = {
+  editingTransactionIndex: number;
+  transaction: ProposedTransaction;
+  onSubmit: (transaction: ProposedTransaction) => void;
+  onDelete: (transactionIndex: number) => void;
+  onClose: () => void;
+  nativeCurrencySymbol: string;
+  networkPrefix: string;
+  getAddressFromDomain: (name: string) => Promise<string>;
+};
+
+const EditTransactionModal = ({
+  editingTransactionIndex,
+  transaction,
+  onClose,
+  nativeCurrencySymbol,
+  networkPrefix,
+  getAddressFromDomain,
+  onSubmit,
+  onDelete,
+}: EditTransactionModalProps) => {
+  const initialFormValues: Partial<SolidityFormValuesTypes> = {
+    [TO_ADDRESS_FIELD_NAME]: transaction.raw.to,
+    [NATIVE_VALUE_FIELD_NAME]: weiToEther(transaction.raw.value),
+    [CUSTOM_TRANSACTION_DATA_FIELD_NAME]: transaction.description.customTransactionData,
+  };
+
+  // If the transaction is not a custom hex encoded data, we need to set the contract method index,
+  // we need to set the contract fields values
+  const contractMethodEditable =
+    !transaction.description.customTransactionData &&
+    transaction.description.contractMethod &&
+    transaction.contractInterface;
+  if (contractMethodEditable) {
+    initialFormValues[CONTRACT_METHOD_INDEX_FIELD_NAME] = transaction.description.contractMethodIndex;
+
+    // if fallback method is used, there are no values, therefore we check if values are present before setting them
+    if (transaction.description.contractFieldsValues) {
+      for (const [paramName, argument] of Object.entries(transaction.description.contractFieldsValues)) {
+        initialFormValues[CONTRACT_VALUES_FIELD_NAME] = {
+          ...initialFormValues[CONTRACT_VALUES_FIELD_NAME],
+          [paramName]: argument,
+        };
+      }
+    }
+  }
+
+  const handleSubmit = (values: SolidityFormValuesTypes) => {
+    onClose();
+
+    const editedTransaction = parseFormToProposedTransaction(values, transaction.contractInterface);
+
+    // keep the id of the transaction
+    onSubmit({ ...editedTransaction, id: transaction.id });
+  };
+
+  return (
+    <GenericModal
+      title={`Transaction ${editingTransactionIndex}`}
+      body={
+        <FormContainer>
+          <SolidityForm
+            id="solidity-contract-form"
+            initialValues={initialFormValues}
+            contract={transaction.contractInterface}
+            nativeCurrencySymbol={nativeCurrencySymbol}
+            networkPrefix={networkPrefix}
+            getAddressFromDomain={getAddressFromDomain}
+            defaultHexDataView={!contractMethodEditable}
+            onSubmit={handleSubmit}
+            showHexToggler={false}
+          >
+            <ButtonContainer>
+              {/* Remove transaction btn */}
+              <Button
+                type="button"
+                size="md"
+                color="error"
```

Comment:
We can add this
```
variant="bordered"
```

To fit 100% with [the new design](https://www.figma.com/file/G7snLqBXWNPANcOj5MhiR7/Transaction-Builder?node-id=1%3A2):
![Captura de pantalla 2022-02-21 a las 14 27 55](https://user-images.githubusercontent.com/26763673/154964444-5c34c282-3ffe-4d77-99c9-14a025143cd1.png)